### PR TITLE
LibHTTP: Support Transfer-Encoding: chunked

### PR DESCRIPTION
**Pull request by alimpfard**
We advertise ourselves to servers as supporting HTTP/1.1; we should put our money where our mouth is, and start supporting some of its features.

p.s. I tried to open nyaa.si/?q=foo
> muh anime torrents

### CODE REVIEW

File: Libraries/LibHTTP/Job.cpp [hunk 1]
```diff
@@ -151,7 +151,50 @@ void Job::on_socket_connected()
...
+                if (remaining == -1) {
+                    // read size
+                    auto size_data = read_line(PAGE_SIZE);
+                    auto size_lines = StringView { size_data.data(), size_data.size() }.lines();
+                    if (size_lines.size() == 0) {
+                        dbg() << "Job: Reached end of stream";
+                        read_line(PAGE_SIZE);
+                    } else {
+                        String size_string = size_lines[0];
+                        auto size = strtoul(size_string.characters(), nullptr, 16);
```
Comment (awesomekling) [inline Libraries/LibHTTP/Job.cpp:167]: What if it's not a valid number?

File: Libraries/LibHTTP/Job.cpp [hunk 2]
```diff
@@ -163,9 +206,37 @@ void Job::on_socket_connected()
...
+                    size = -1;
+                    auto line = read_line(PAGE_SIZE);
+#ifdef JOB_DEBUG
+                    dbg() << "Line following (should be empty): _" << line << "_";
+#endif
+                    (void)line;
+
+                    if (m_current_chunk_total_size.value() == 0) {
+                        // this was the last chunk, read the rest and discard it.
+                        (void)receive(64 * KB);
```
Comment (awesomekling) [inline Libraries/LibHTTP/Job.cpp:232]: Why is 64 KB the right max number of bytes to receive to "read the rest"?

File: Libraries/LibHTTP/Job.cpp (hunk 1 above)
Comment (alimpfard) [reply Libraries/LibHTTP/Job.cpp:167]: hmmm...I suppose we should fail then.

now how does that endptr work...?

File: Libraries/LibHTTP/Job.cpp (hunk 2 above)
Comment (alimpfard) [reply Libraries/LibHTTP/Job.cpp:232]: I figured it's big enough for a single chunk.
But this actually leads into another problem I just discovered, chunked encoding allows _more_ headers after the last chunk

File: Libraries/LibHTTP/Job.cpp (hunk 1 above)
Comment (alimpfard) [reply Libraries/LibHTTP/Job.cpp:167]: > how does that endptr work

It actually doesn't.
I'll add the fix to this PR, if that's alright?

Comment (awesomekling) [reply Libraries/LibHTTP/Job.cpp]: Ofc!

File: Libraries/LibC/stdlib.cpp [hunk 3]
```diff
@@ -1003,6 +1003,9 @@ unsigned long long strtoull(const char* str, char** endptr, int base)
         return 0;
     }
 
+    if (endptr)
+        *endptr = parse_ptr;
```
Comment (awesomekling) [inline Libraries/LibC/stdlib.cpp:1007]: How about we also fix `strtoll()` while we're here :]

**Pull request merged**